### ISSUE TITLE
Hydration causing large CLS

### ISSUE THREAD

**Issue by chemeng** [brillout/vite-plugin-ssr#305]
Thanks for the great work here. I'm using less (ant-design framework) in my vue project and with the library I get some pretty large cumulative layout shifts due to javascript not preloading. Is there a way to force the client javascript file (`/assets/src/all.page.client.js`) to be included in the head instead of bottom of the body section similar to the `renderResourceHints()` of the bundle renderer?

**Comment by brillout** [comment]
Yes it's on the roadmap, see https://github.com/brillout/vite-plugin-ssr/issues/262.

It may already be released in `0.4`. No guarantees though. In the meantime, it should be fairly easy to patch vite-plugin-ssr to do what you want, see `injectAssets.ts`.

Closing this in favor of https://github.com/brillout/vite-plugin-ssr/issues/306.

**Comment by chemeng** [comment]
Thank you very much for your reply and all the work here.

I'm actually not sure that the javascript injection I mentioned earlier was the problem for me after all. The problem  appears to be that the `css` of each vue component is not included in the client css bundle and is only loaded dynamically during rendering causing the large shifts. 

If i simply adjust the `manifest.json` of the client and include the css files for my components in the `src/all/page.client.js` entry everything works fine. Is there a way to "bundle" those css files that are needed with the `all.page.client.js.css` file that is created? 

For now and as a workaround I explicitly imported all the component css files I need before rendering in my `app.js` script that creates the vue app.

Btw, I'm using vue-router for client-based navigation btw (not sure if that is a reason for the dynamic dependencies not being included). Finally, since I wasn't able to find it in the docs, what's the difference between `_default.page.client.js` and `all.page.client.js` ?

**Comment by brillout** [comment]
I recommend using the built-in client-side router instead of Vue Router. That will probably fix the issue. (I suspect that Vue Router dynamically loads components inducing a delay for the CSS loading.)

**Comment by brillout** [comment]
I'm curious; why do you want to use Vue Router in the first place?

**Comment by chemeng** [comment]
To be honest, I've been using vue-router in non-ssr projects for a few years now so I know it I guess. 

Additionally, I use the `this.$route.path`, url params parsing, routing using the name instead of the path, etc. I'm pretty sure I can do these things without vue-router but I'm not sure how yet.

**Comment by brillout** [comment]
Yes you can also do that with vps's Client Routing.

If you don't have a strong reason to use vue-router, then don't.

**Comment by chemeng** [comment]
Is there an easy way to replicate the `<router-view>` functionality? For example, in vue-router I have a "layout" component at the root route and the rest are children.

```
<template>
  <div id="root">
      <layout-header />
        <router-view />
      <layout-footer />
  </div>
</template>
```

**Comment by brillout** [comment]
Yes you can do that.

You can also do nested routing. I'll implement an example with vps 0.4 and show you when it's ready.

**Comment by chemeng** [comment]
thank you, much appreciated. what's your timeline on the 0.4 release?

**Comment by brillout** [comment]
This/next week.

> I'll implement an example with vps 0.4 and show you when it's ready.

Follow https://github.com/brillout/vite-plugin-ssr/issues/309 to get updates.

**Comment by chemeng** [comment]
hey, is that example somewhere that I can take a look at? very excited for the 0.4 release

**Comment by brillout** [comment]
Not yet but ping me again next week.

**Comment by chemeng** [comment]
Pinging again for that example.

**Comment by brillout** [comment]
I will publish an example today. Thanks for the ping.

**Comment by brillout** [comment]
https://github.com/brillout/vite-plugin-ssr/tree/0.4.0/examples/layouts-react

See the `/starship` page which features a nested layout.

I'll write a Vue example tomorrow. The idea will be the same.

**Comment by brillout** [comment]
Vue example: https://github.com/brillout/vite-plugin-ssr/tree/0.4.0/examples/layouts-vue

**Comment by chemeng** [comment]
thank you, this is available with 0.4 only right? (not with current prod version)

**Comment by brillout** [comment]
Correct
